src/components/QueryTimeslotTimePicker.tsx

```diff
@@ -47,6 +50,15 @@ export default function QueryTimeslotTimePicker(
     newDate.setHours(value?.getHours() ?? 0);
     newDate.setMinutes(value?.getMinutes() ?? 0);
 
+    const diff = (newDate.getTime() - new Date().getTime()) / 3600000;
```

Comment:
Das hier musst du in einem useEffect machen, der an der query hängt. Sonst funktioniert es nich, wenn man das Datum im Kalender ändert.